app/src/main/java/no/usn/mob3000/domain/viewmodel/ProfileViewModel.kt

```diff
@@ -0,0 +1,37 @@
+package no.usn.mob3000.domain.viewmodel
+
+/**
+ * @author Hussein Abdul-Ameer
+ * created on : 01.11.2024
+ */
+import android.util.Log
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.launch
+import no.usn.mob3000.domain.model.Friend
+import no.usn.mob3000.domain.usecase.social.FetchFriendsUseCase
+
+class ProfileViewModel(
+    private val fetchFriendsUseCase: FetchFriendsUseCase = FetchFriendsUseCase()
+) : ViewModel() {
+
+    private val _friends = MutableStateFlow<List<Friend>>(emptyList())
+    val friends: StateFlow<List<Friend>> get() = _friends
+
+    init {
+        loadFriends()
+    }
```

Comment:
Inits in ViewModels are very often not necessary. Consider if the structure actually needs this, or if it was just the simpler option, where there is a better option elsewhere. E.g. for making Unit testing easier, etc.